### ISSUE TITLE
[.5] Clear fields once transfer operation is complete

### ISSUE THREAD

**Issue by wmbutler** [bitshares/bitshares-ui#533]
Once a transfer completes, it is important for the UI to clear the following fields:

- [ ] To
- [ ] Quantity
- [ ] Memo / Message

![transfer](https://user-images.githubusercontent.com/1254810/31358850-f5800c62-ad0c-11e7-8994-8fb42d7e57c7.gif)

**Comment by tbone-bts** [comment]
Sometimes it's actually very handy (and potentially less dangerous) not having to retype and/or copy/paste the same address and memo again when you send back-to-back transfers to the same recipient.  But I can also see where you are coming from on this.  Perhaps there could be a checkbox for the user to tick off if they don't want the fields to clear after sending (e.g. "[  ] Don't clear form values after sending.").  Perhaps even better would be some sort of address book.

**Comment by wmbutler** [comment]
If they really want the address, they can copy it to their clipboard. I agree that we need an address book, but why would you want the same username in the to field? You just sent them funds... Why would you pay someone a dollar and then a dollar. Just give them two dollars, once.

**Comment by landry314** [comment]
Fields should be cleared, definitely, it is cleaner that way.

Also, a mistake could be made if you are sending to a few different accounts in a row and you absentmindedly forget to change the Quantity field, it could be a big mistake.

**Comment by tbone-bts** [comment]
Hey Bill, I'm not suggesting sending to the same recipient immediately.  But if I want/need to make recurring payments (which is a thing, and it's actually a feature of Bitshares, just not supported in the GUI), on a daily or weekly basis, then I can keep a browser tab open with the persisting details and use it to send multiple payments without having to both 1) copy and paste the **address** again and 2) copy and paste the **memo** again, which can be error prone and possibly lead to loss of funds (and believe me I've seen it happen).

Anyway, the landry guy who posted after you didn't seem to understand that I was not suggesting to have the details persist by default.  I was merely suggesting to add a check box that is **unchecked** by default but can be checked if the user wants to persist the payment details in order to send multiple payments to the same recipient.  Short of implementing an address book, this is a very easy way to accommodate all needs and avoid removing a functionality that some have become accustomed to.  

>If they really want the address, they can copy it to their clipboard. I agree that we need an address book, but why would you want the same username in the to field? You just sent them funds... Why would you pay someone a dollar and then a dollar. Just give them two dollars, once.

**Comment by wmbutler** [comment]
@tbone-bts I get it, but your suggestion is really a feature request. The goal of this issue it to ensure that bad things don't happen by leaving previously used forms populated.

**Comment by tbone-bts** [comment]
I guess it's a feature request.  But really an existing feature is being taken away and I'm just suggesting to do it in a more elegant way so as not to alienate those who appreciate the existing feature.  

P.S. Most of the time people will make a single transfer and navigate away from the page.  It's not often people will make multiple back-to-back transfers, although the need can and does surely arise once in a while (even though it never has for me).  By the same token, the need I described previously absolutely also arises.  Anyway, so one of these exceptions is already accommodated.  Now that accommodation will be removed to accommodate the other exception.  It doesn't seem logical to do that, especially when a simple check box can enable both to be accommodated.

**Comment by landry314** [comment]
But it is not removing a feature, it is fixing an oversight in the code.

To add the feature of reoccurring payments, why not create a new issue now for exactly this? If there is a much cleaner and expected way to do things that will not inadvertently cause mistakes for some people, then let's do that!

**Comment by tbone-bts** [comment]
You call it "fixing an oversight" because you're biased with your own point of view.  Whereas I am looking at it form BOTH sides.  If I wasn't, I would be advocating for leaving it as is.  But that's not what I'm doing. 
 By the way, why do you have a problem with the idea of adding a checkbox that let's people override the new default behavior such that everyone can be accommodated?  Yes, of course recurring payments would be nice and would help in this regard.  And so would an address book.  But those features require far more work and are not high priorities.  Whereas a simple check box would take practically no time and would be a good interim solution.  That's called a "no-brainer".

**Comment by wmbutler** [comment]
Because you are requesting a feature. Forms that have been used are intended to be reset after use.
